src/EditorFeatures/Core/LanguageServer/AlwaysActivateInProcLanguageClient.cs

```diff
@@ -95,6 +101,10 @@ public override ServerCapabilities GetCapabilities(ClientCapabilities clientCapa
                     new(PullDiagnosticCategories.DocumentAnalyzerSyntax),
                     new(PullDiagnosticCategories.DocumentAnalyzerSemantic),
                 };
+                serverCapabilities.DiagnosticProvider.BuildOnlyDiagnosticIds = _buildOnlyDiagnostics
+                    .SelectMany(lazy => lazy.Metadata.BuildOnlyDiagnostics)
+                    .Distinct()
+                    .ToArray();
```

Comment:
by accessing metadata only, we can pull in the info from C#/VB without unnecessary dll loads.

Reply:
I *think* this should work, but we should run an RPS val build to be certain - build here - https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=7169548&view=results